controllers/model/grafanaDataPvc.go

```diff
@@ -32,6 +32,14 @@ func getStorageClass(cr *v1alpha1.Grafana) *string {
 	return &cr.Spec.DataStorage.Class
 }
 
+func getVolumeName(cr *v1alpha1.Grafana) string {
+	if cr.Spec.DataStorage == nil {
```

Comment:
Hard to review on phone but could it be reasonable to also check for an empty string just go make sure that people don't to that mistake?

Other then that i think it looks good.

Reply:
I think a check for empty string is sufficient here

Reply:
`cr.Spec.DataStorage` is not of type string.
And checking `VolumeName` for an empty string doesn't make sense, as the default return-value is (and must be) an empty string anyway.

Reply:
Yes you are right. Me and @pb82 trying to do to many things at the same time.